boot/bootutil/src/loader.c

```diff
@@ -2868,12 +2868,14 @@ boot_load_and_validate_images(struct boot_loader_state *state,
  *
  * @param  state        Boot loader status information.
  * @param  active_slot  Index of the slot will be loaded for current image.
+ * @param  slot_usage   Information about the active and available slots.
  *
  * @return              0 on success; nonzero on failure.
  */
 static int
 boot_update_hw_rollback_protection(struct boot_loader_state *state,
-                                   uint32_t active_slot)
+                                   uint32_t active_slot,
+                                   const struct slot_usage_t slot_usage[])
```

Comment:
Please remove the `active_slot` from the parameter list and extract the active slot index from the the `slot_usage` array inside function.

Reply:
Done